realm/realm-library/src/main/java/io/realm/BaseRealm.java

```diff
@@ -282,6 +284,47 @@ public void refresh() {
     }
 
     /**
+     * Blocks the current thread until new changes to the Realm are available or {@link #stopWaitForChange()}
+     * is called from another thread. Once stopWaitForChange is called, all future calls to this will
+     * return false immediately.
+     *
+     * @return {@code true} If the Realm was updated to the latest version, {@code false} if it was
+     * cancelled by calling stopWaitForChange.
+     * @throws IllegalStateException if calling this from within a transaction or from a Looper thread.
+     */
+    public boolean waitForChange() {
+        checkIfValid();
+        if (isInTransaction()) {
+            throw new IllegalStateException("Cannot wait for changes inside of a transaction.");
+        }
+        if (Looper.myLooper() != null) {
+            throw new IllegalStateException("Cannot wait for changes inside a Looper thread. Use RealmChangeListeners instead.");
+        }
+        boolean hasChanged = sharedGroupManager.getSharedGroup().waitForChange();
+        if (hasChanged) {
+            // Since this Realm instance has been waiting for change, advance realm & refresh realm.
+            sharedGroupManager.advanceRead();
+            handlerController.refreshSynchronousTableViews();
+        }
+        return hasChanged;
+    }
+
+    /**
+     * Makes any current {@link #waitForChange()} return {@code false} immediately. If a Realm is not
+     * waiting, all future calls to waitForChange will immediately return false as well.
+     * <p>
+     * This method is threadsafe and should _only_ be called from another thread than the one that
+     * called waitForChange.
+     */
+    public void stopWaitForChange() {
```

Comment:
I had the same discussion with @nhachicha , we talked about `shutdown` and `exit` because he had the same reservations as you do. I think `disable` suffers from the same as `stop`, namely that you wonder where the `enableWaitForChange()` is then.

I think we ended up with keeping `stopWaitForChange()`, mostly because there is a non-zero chance that core will change their API, so we can actually calling `wait/stop/wait/stop` multiple times. If that happens, we don't have to change the public API.

IMO `disable` doesn't sound like something that would effect any current `wait` either, but I agree, there none of the options are really _great_.